### PR TITLE
Update exporter-toolkit

### PR DESCRIPTION
**Pull request by SuperQ**
* Update Go to 1.23.
* Replace go-kit/log with slog.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -401,12 +396,12 @@ func main() {
...
 		tcpListenAddr, err := address.TCPAddrFromString(*statsdListenTCP)
 		if err != nil {
-			level.Error(logger).Log("msg", "invalid TCP listen address", "address", *statsdListenUDP, "error", err)
+			logger.Error("invalid TCP listen address", "address", *statsdListenUDP, "error", err)
 			os.Exit(1)
 		}
 		tconn, err := net.ListenTCP("tcp", tcpListenAddr)
 		if err != nil {
-			level.Error(logger).Log("msg", err)
+			logger.Error(err.Error())
```
Comment (tjhop) [inline main.go:404]:
```suggestion
			logger.Error("failed to start TCP listener", "err", err)
```

File: main.go [hunk 2]
```diff
@@ -507,7 +502,7 @@ func main() {
 		}
 		landingPage, err := web.NewLandingPage(landingConfig)
 		if err != nil {
-			level.Error(logger).Log("err", err)
+			logger.Error(err.Error())
```
Comment (tjhop) [inline main.go:505]:
```suggestion
			logger.Error("error creating landing page", "err", err)
```

File: pkg/listener/listener.go [hunk 3]
```diff
@@ -64,7 +63,7 @@ func (l *StatsDUDPListener) Listen() {
 			if strings.HasSuffix(err.Error(), "use of closed network connection") {
 				return
 			}
-			level.Error(l.Logger).Log("error", err)
+			l.Logger.Error(err.Error())
```
Comment (tjhop) [inline pkg/listener/listener.go:66]:
```suggestion
			l.Logger.Error("error reading from UDP connection", "err", err)
```

File: pkg/listener/listener.go [hunk 4]
```diff
@@ -198,7 +197,7 @@ func (l *StatsDUnixgramListener) Listen() {
 			if strings.HasSuffix(err.Error(), "use of closed network connection") {
 				return
 			}
-			level.Error(l.Logger).Log(err)
+			l.Logger.Error(err.Error())
```
Comment (tjhop) [inline pkg/listener/listener.go:200]:
```suggestion
			l.Logger.Error("error reading from unixgram connection", "err", err)
```

**Review by tjhop — APPROVED** [review]
left a few suggestions to improve "naked" errors, but looks great :+1:

**Comment by SuperQ** [comment]
Incorporated suggestions.

**Review by tjhop — APPROVED** [review]
❤️

**Comment by tjhop** [comment]
@matthiasr is there any chance you could take a look at this sometime? The graphite_exporter imports the statsd_exporter to use the `mapper.MetricMapping` struct, so it seems this is a prereq to updating the graphite_exporter to fully use log/slog

**Review by matthiasr — APPROVED** [review]

**Comment by matthiasr** [comment]
Thank you both 🙏 🙏 🙏

**Pull request merged**